tests/fullydistributed_grids/repartitioning_05.cc

```diff
@@ -109,7 +109,7 @@ test(const MPI_Comm comm, const unsigned int n_partitions)
 
   parallel::distributed::Triangulation<dim> tria(sub_comm);
 
-  if (sub_comm != MPI_COMM_NULL)
+  if (sub_comm != MPI_COMM_SELF)
```

Comment:
Thanks for tackling this! 

I am not sure if this is equivalent to the previous code: this only works if `MPI_COMM_WORLD` and `MPI_COMM_SELF` are not the same (and if a subcommunicator of size one is not equivalent to `MPI_COMM_SELF`) for a serial execution. Is this the case?